### PR TITLE
Data model

### PR DESCRIPTION
**Pull request by Beardless-sheik**
- Project requirements

  - Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
  - Set up associations between models.
Your models should include some additional methods.

- Users

  - A method that returns the 3 most recent posts for a given user.

- Posts
  - A method that updates the posts counter for a user.
  - A method which returns the 5 most recent comments for a given post.

- Comments
  - A method that updates the comments counter for a post.

- Likes
  - A method that updates the likes counter for a post.
  

- Create a current_user method in your ApplicationController.

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post, counter_cache: :comments_count
+
+  def update_count(count)
+    post.uodate('comments_count', count)
```
Comment (ahmetbozaci) [inline app/models/comment.rb:6]: - [ ] It should be`  post.increment!(:comments_counter)
`. Please fix it.

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post, counter_cache: :likes_count
+
+  def update_count(count)
+    post.update(likes_count: count)
+  end
```
Comment (ahmetbozaci) [inline app/models/like.rb:7]: - [ ] It should be `post.increment!(:likes_counter)`. Please fix it.

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  belongs_to :user, class_name: 'User', counter_cache: :posts_count
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def recent_comments
+    comments.last(5)
+  end
+
+  def update_count(count)
+    user.update('posts_count', count)
```
Comment (ahmetbozaci) [inline app/models/post.rb:11]: - [ ] It should be `author.increment!(:posts_counter)`. Please fix it.

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️


 @Beardless-sheik  ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.

### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._
- [ ] You should add the first user method in the application.rb file.  [Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md#:~:text=A%20current_user%20method%20should%20return%20the%20first%20user%20from%20a%20database.)

image
___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
- [ ]  You have some mistakes in the schema from the last milestone. You made some typo when writing foreign_key. 
- [ ] Also column names should match the picture in the requirement.
___

### Cheers and Happy coding! 💻 👏
almost
almost


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Comment by Beardless-sheik** [comment]
Thanks for the recommendations. Working on it.

**Review by EricMbouwe — APPROVED** [review]
Hi @Beardless-sheik, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Beardless-sheik** [comment]
> Hi @Beardless-sheik,
> 
> Your project is complete! There is nothing else to say other than... it's time to merge it :shipit: Congratulations! tada
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
> 
> Cheers and Happy coding!clapclapclap
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Thanks.......

**Pull request merged**